pkg/console/operator/status.go

```diff
@@ -57,6 +58,59 @@ const (
 	reasonCustomLogoInvalid   = "CustomLogoInvalid"
 )
 
+// handleDegraded can be used to set a number of Degraded statuses
+// example:
+//   co.handleDegraded(operatorConfig, "RouteStatus", err)
+//   co.handleDegraded(operatorConfig, "CustomLogoInvalid", err)
+// creates condition:
+//   RouteStatusDegraded
+//   CustomLogoInvalidDegraded
+// and uses the error as its message
+func (c *consoleOperator) HandleDegraded(operatorConfig *operatorsv1.Console, prefix string, err error) {
+	conditionType := prefix + operatorsv1.OperatorStatusTypeDegraded
+	reason := prefix + "Error"
+	handleCondition(operatorConfig, conditionType, reason, err)
+}
+
+func (c *consoleOperator) HandleProgressing(operatorConfig *operatorsv1.Console, prefix string, err error) {
+	conditionType := prefix + operatorsv1.OperatorStatusTypeProgressing
+	handleCondition(operatorConfig, conditionType, prefix, err)
+}
+
+func (c *consoleOperator) HandleAvailable(operatorConfig *operatorsv1.Console, prefix string, err error) {
+	conditionType := prefix + operatorsv1.OperatorStatusTypeAvailable
+	handleCondition(operatorConfig, conditionType, prefix, err)
+}
+
+// internal func for handling conditions
+func handleCondition(operatorConfig *operatorsv1.Console, conditionType string, reason string, err error) {
+	if err != nil {
+		v1helpers.SetOperatorCondition(&operatorConfig.Status.Conditions,
+			operatorsv1.OperatorCondition{
+				Type:    conditionType,
+				Status:  operatorsv1.ConditionTrue,
+				Reason:  reason,
+				Message: err.Error(),
+			})
+		return
+	}
+	v1helpers.SetOperatorCondition(&operatorConfig.Status.Conditions,
+		operatorsv1.OperatorCondition{
+			Type:   conditionType,
+			Status: operatorsv1.ConditionFalse,
+		})
+}
+
+func IsDegraded(operatorConfig *operatorsv1.Authentication) bool {
```

Comment:
Can't find where this func is used

Reply:
Neither are the `HandleProgressing()` and `HandleAvailable()`, but I put all these helpers in this first PR so I can build on them.